src/backend/joanie/tests/core/test_utils_issuers_contract_definition_generate_document.py

```diff
@@ -17,39 +18,235 @@ class UtilsIssuersContractDefinitionGenerateDocument(TestCase):
     format.
     """
 
-    def test_utils_contract_definition_generate_document(self):
+    def test_utils_issuers_contract_definition_generate_document(self):
         """
         Issuer 'generate document' method should generate a contract definition document.
         """
+        factories.SiteConfigFactory(
+            site=Site.objects.get_current(),
+            terms_and_conditions="Terms and Conditions Content",
+        )
+
         user = factories.UserFactory(
-            email="student@example.fr", first_name="John", last_name="Doe"
+            email="student@example.fr",
+            first_name="Rooky",
+            last_name="The Student",
+            phone_number="0612345678",
         )
-        address = factories.UserAddressFactory(
-            owner=user,
-            address="1 Rue de L'Exemple",
-            postcode="75000",
-            city="Paris",
-            is_reusable=False,
-            title="Office",
+
+        definition = factories.ContractDefinitionFactory(
+            title="Contract Definition Title",
+            description="Contract Definition Description",
+            body="## Contract Definition Body",
+        )
+
+        organization = factories.OrganizationFactory(
+            title="University X",
+            activity_category_code="8542Z",
+            enterprise_code="UX-74392-899",
+            representative="John Doe",
+            representative_profession="Director",
+            signatory_representative="Jane Doe",
+            signatory_representative_profession="Administrative Manager",
+            contact_phone="+33712345678",
+            contact_email="contact@university-x.xyz",
+            dpo_email="dpo@university-x.xyz",
+        )
+
+        factories.OrganizationAddressFactory(
+            organization=organization,
+            address="1 Rue de l'Université",
+            postcode="87000",
+            city="Limoges",
             country="FR",
+            is_reusable=True,
+            is_main=True,
+        )
+
+        product = factories.ProductFactory(
+            title="You will know that you know you don't know",
```

Comment:
`Win the “yes” needs the “no” to win against the “no”`

_this is not a suggestion_